firstvoices/backend/search/indices/dictionary_entry_document.py

```diff
@@ -271,14 +274,25 @@ def update_acknowledgement(sender, instance, **kwargs):
         logger.error(e)
 
 
-# Category update
+# Category update when called through the admin site
 @receiver(post_save, sender=DictionaryEntryCategory)
 @receiver(post_delete, sender=DictionaryEntryCategory)
 def update_categories(sender, instance, **kwargs):
-    logger = logging.getLogger(ELASTICSEARCH_LOGGER)
     dictionary_entry = instance.dictionary_entry
-    categories = get_categories_ids(dictionary_entry)
+    update_dictionary_entry_index_categories(dictionary_entry, instance)
+
+
+# Category update when called through the APIs
+@receiver(m2m_changed, sender=DictionaryEntryCategory)
+def update_categories_m2m(sender, instance, **kwargs):
+    if instance.__class__ == DictionaryEntry:
+        dictionary_entry = instance
+        update_dictionary_entry_index_categories(dictionary_entry, instance)
 
+
+def update_dictionary_entry_index_categories(dictionary_entry, instance):
```

Comment:
Why do we need both these parameters, if they are the same?

Reply:
Good catch. I have updated this.